plugin.video.netzkino_de/resources/lib/kodion/impl/xbmc/xbmc_runner.py

```diff
@@ -0,0 +1,151 @@
+__author__ = 'bromix'
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+
+from ..abstract_provider_runner import AbstractProviderRunner
+from ...exceptions import KodionException
+from ...items import *
+from ... import constants, AbstractProvider
+from . import info_labels
+from . import xbmc_items
+
+
+class XbmcRunner(AbstractProviderRunner):
+    def __init__(self):
+        AbstractProviderRunner.__init__(self)
+        pass
+
+    def run(self, provider, context=None):
+        results = None
+        try:
+            results = provider.navigate(context)
+        except KodionException, ex:
+            if provider.handle_exception(context, ex):
+                context.log_error(ex.__str__())
+                xbmcgui.Dialog().ok("Exception in ContentProvider", ex.__str__())
+                pass
+            return
+
+        result = results[0]
+        options = {}
+        options.update(results[1])
+
+        if isinstance(result, bool) and not result:
+            xbmcplugin.endOfDirectory(context.get_handle(), succeeded=False)
+        elif isinstance(result, VideoItem) or isinstance(result, AudioItem) or isinstance(result, UriItem):
+            self._set_resolved_url(context, result)
+        elif isinstance(result, DirectoryItem):
+            self._add_directory(context, result)
+        elif isinstance(result, list):
+            item_count = len(result)
+            for item in result:
+                if isinstance(item, DirectoryItem):
+                    self._add_directory(context, item, item_count)
+                elif isinstance(item, VideoItem):
+                    self._add_video(context, item, item_count)
+                elif isinstance(item, AudioItem):
+                    self._add_audio(context, item, item_count)
+                elif isinstance(item, ImageItem):
+                    self._add_image(context, item, item_count)
+                pass
+
+            xbmcplugin.endOfDirectory(
+                context.get_handle(), succeeded=True,
+                cacheToDisc=options.get(AbstractProvider.RESULT_CACHE_TO_DISC, True))
+
+            # set alternative view mode
```

Comment:
This kind of stuff is not allowed anymore in the repo. Plugins should not force views but instead correctly define the content they provide. Having this built like that forces you to keep track of every skin id's and versions and may easily lead to errors. Also, it forces users to define view modes in more than one place...which turns out to be a terrible user-experience.